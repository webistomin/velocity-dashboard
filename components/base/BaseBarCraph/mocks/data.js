export const barGraphData = {
  labels: ['April', 'May', 'June'],
  datasets: [
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      lineTension: 0.4,
      backgroundColor: '#2E5BFF',
      data: [60, 65, 85],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [70, 80, 90],
    },
  ],
};
