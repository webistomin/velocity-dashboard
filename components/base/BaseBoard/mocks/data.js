export const boardData = [
  {
    title: 'Backlog',
    color: 'blue',
    tasks: [
      {
        title: 'Spire1',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire2',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire3',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'In progress',
    color: 'purple',
    tasks: [
      {
        title: 'Spire5',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire6',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire7',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire8',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Validate',
    color: 'darkturquoise',
    tasks: [
      {
        title: 'Spire9',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire10',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire11',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire12',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Complete',
    color: 'yellow',
    tasks: [
      {
        title: 'Spire13',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire14',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire15',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire16',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
];
