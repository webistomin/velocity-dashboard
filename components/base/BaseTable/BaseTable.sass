+b('base-table')
  +e('scroller')
    +scrollbar(5px, var(--color-primary), var(--color-secondary))
    height: calc(100vh - 350px)

    .vue-recycle-scroller__slot
      padding-top: rem(10px)
      padding-bottom: rem(10px)
      position: sticky
      top: 0
      z-index: 1

    .vue-recycle-scroller__item-wrapper
      overflow-x: auto
      overflow-y: hidden

  +e('table')
    position: relative
    width: 100%
    border-spacing: 0 rem(10px)

  +e('row')
    display: flex
    background-color: var(--color-white)

  +e('head')
    display: flex
    padding-bottom: rem(10px)
    overflow-x: auto
    overflow-y: hidden

  +e('head-data')
    padding-right: rem(24px)

    &:first-of-type
      padding-left: rem(24px)

  +e('sort')
    display: flex
    align-items: center
    transition: color 0.2s ease-out

    &:hover
      color: var(--color-primary)

  +e('sort-name')
    margin-right: rem(10px)

  +e('body-data')
    box-sizing: border-box
    display: inline-block
    padding-top: rem(30px)
    padding-bottom: rem(30px)
    padding-right: rem(24px)

    &:first-of-type
      padding-left: rem(24px)

  +e('cell')
    flex: 23 1 0

    &:first-of-type
      flex: 8 1 0
