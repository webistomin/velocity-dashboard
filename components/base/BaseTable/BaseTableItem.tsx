import Vue, { RenderContext, VNode, CreateElement } from 'vue';

export interface IBaseTableItemProps {
  index: number;
  source: object;
}

export const BaseTableItem = Vue.extend({
  functional: true,
  props: {
    index: {
      type: Number,
      default: 0,
    },
    source: {
      type: Object,
      default: () => {},
    },
  },
  render(_h: CreateElement, ctx: RenderContext<IBaseTableItemProps>): VNode {
    const { staticClass, class: cls } = ctx.data;
    const { source } = ctx.props;
    return (
      <div class={`base-table__row ${staticClass || ''} ${cls || ''}`}>
        {Object.values(source).map((field) => {
          return <span class='base-table__body-data base-table__cell'>{field}</span>;
        })}
      </div>
    );
  },
});
