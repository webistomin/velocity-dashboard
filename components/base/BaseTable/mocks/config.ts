export const config = [
  {
    key: 'id',
    title: 'Id',
  },
  {
    key: 'manufacturer',
    title: 'Manufacturer',
  },
  {
    key: 'model',
    title: 'Model',
  },
  {
    key: 'location',
    title: 'Location',
  },
  {
    key: 'mileage',
    title: 'Mileage',
  },
  {
    key: 'date',
    title: 'Date of purchase',
  },
  {
    key: 'status',
    title: 'Status',
  },
  {
    key: 'tripsTaken',
    title: 'Trips taken',
  },
];
