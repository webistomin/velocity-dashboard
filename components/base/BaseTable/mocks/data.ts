export const data = [
  {
    id: 'Niuq1',
    manufacturer: 'Mercedes-Benz',
    model: 'C1',
    location: 'Texas, TX',
    mileage: 97856,
    dateOfPurchase: '2006-03-03T21:00:00.000Z',
    status: 'fully serviced',
    tripsTaken: 541,
  },
  {
    id: 'bTArp',
    manufacturer: 'BMW',
    model: 'A2',
    location: 'Washington DC, DC',
    mileage: 24412,
    dateOfPurchase: '2008-04-30T20:00:00.000Z',
    status: 'waiting',
    tripsTaken: 761,
  },
  {
    id: 'bwnVE',
    manufacturer: 'Ford',
    model: 'B1',
    location: 'Texas, TX',
    mileage: 14475,
    dateOfPurchase: '2002-02-04T21:00:00.000Z',
    status: 'service needed',
    tripsTaken: 958,
  },
  {
    id: 'mMTzR',
    manufacturer: 'Mazda',
    model: 'C2',
    location: 'Washington DC, DC',
    mileage: 72493,
    dateOfPurchase: '2005-03-06T21:00:00.000Z',
    status: 'service needed',
    tripsTaken: 176,
  },
];
