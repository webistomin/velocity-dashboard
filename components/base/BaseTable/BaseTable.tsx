import { VueComponent } from 'types/vue-components';
import { Component, Prop } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import './BaseTable.sass';

export interface IBaseTableConfig {
  key: string;
  title: string;
}

export interface IBaseTableProps {
  tableConfig: IBaseTableConfig[];
  tableData: object[];
}

@Component({
  name: 'BaseTable',
})
export default class BaseTable extends VueComponent<IBaseTableProps> {
  @Prop()
  private readonly tableConfig!: IBaseTableProps['tableConfig'];

  @Prop()
  private readonly tableData!: IBaseTableProps['tableData'];

  onSortClick(_field: IBaseTableConfig['key']) {}

  get getTableRows() {
    return this.tableData.map((row) => {
      return (
        <div class='base-table__row'>
          {Object.values(row).map((field) => {
            return <span class='base-table__body-data base-table__cell'>{field}</span>;
          })}
        </div>
      );
    });
  }

  public render(): VNode {
    return (
      <div class='base-table'>
        <div class='base-table__wrapper'>
          <div class='base-table__head'>
            {this.tableConfig.map((config) => {
              return (
                <div class='base-table__head-data base-table__cell'>
                  <button
                    class='base-table__sort btn caption'
                    type='button'
                    onClick={() => this.onSortClick(config.key)}>
                    <span class='base-table__sort-name'>{config.title}</span>
                    <svg-icon class='base-table__icon' name='icon-dropdown' width={16} height={16} />
                  </button>
                </div>
              );
            })}
          </div>
          <DynamicScroller
            class='base-table__scroller'
            items={this.tableData}
            item-size={95}
            minItemSize={75}
            key-field='id'>
            <div class='base-table__row'>
              <span class='base-table__body-data base-table__cell'>#2178</span>
              <span class='base-table__body-data base-table__cell'>Refund request</span>
              <span class='base-table__body-data base-table__cell'>05/04/2018</span>
              <span class='base-table__body-data base-table__cell'>Active</span>
            </div>
          </DynamicScroller>
        </div>
      </div>
    );
  }
}
