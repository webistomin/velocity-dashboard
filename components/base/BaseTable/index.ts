import BaseTable from './BaseTable';

export default BaseTable;
