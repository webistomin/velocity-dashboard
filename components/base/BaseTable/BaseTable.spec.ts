import { mount } from '@vue/test-utils';
import BaseTable from './BaseTable';
import { data } from './mocks/data';
import { config } from './mocks/config';

describe('BaseTable', () => {
  const sortSelector = '.base-table__sort';

  const factory = (propsData = {}, options = {}) => {
    return mount(BaseTable, {
      propsData: {
        tableData: data,
        tableConfig: config,
        ...propsData,
      },
      stubs: ['VirtualList', 'svg-icon'],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Emit "sort" event', () => {
    const wrapper = factory();
    const btn = wrapper.find(sortSelector);
    btn.trigger('click');
    expect(wrapper.emitted().sort);
  });
});
