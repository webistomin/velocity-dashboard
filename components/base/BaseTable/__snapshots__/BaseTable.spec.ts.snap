// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTable Matches snapshot 1`] = `
"<div class=\\"base-table\\">
  <div class=\\"base-table__wrapper\\">
    <div class=\\"base-table__head\\">
      <div class=\\"base-table__head-data base-table__cell\\"><button type=\\"button\\" class=\\"base-table__sort btn caption \\"><span class=\\"base-table__sort-name\\">Id</span>
          <svg-icon-stub name=\\"icon-dropdown\\" width=\\"16\\" height=\\"16\\" class=\\"base-table__icon\\"></svg-icon-stub></button></div>
      <div class=\\"base-table__head-data base-table__cell\\"><button type=\\"button\\" class=\\"base-table__sort btn caption \\"><span class=\\"base-table__sort-name\\">Manufacturer</span>
          <svg-icon-stub name=\\"icon-dropdown\\" width=\\"16\\" height=\\"16\\" class=\\"base-table__icon\\"></svg-icon-stub></button></div>
      <div class=\\"base-table__head-data base-table__cell\\"><button type=\\"button\\" class=\\"base-table__sort btn caption \\"><span class=\\"base-table__sort-name\\">Model</span>
          <svg-icon-stub name=\\"icon-dropdown\\" width=\\"16\\" height=\\"16\\" class=\\"base-table__icon\\"></svg-icon-stub></button></div>
      <div class=\\"base-table__head-data base-table__cell\\"><button type=\\"button\\" class=\\"base-table__sort btn caption \\"><span class=\\"base-table__sort-name\\">Location</span>
          <svg-icon-stub name=\\"icon-dropdown\\" width=\\"16\\" height=\\"16\\" class=\\"base-table__icon\\"></svg-icon-stub></button></div>
      <div class=\\"base-table__head-data base-table__cell\\"><button type=\\"button\\" class=\\"base-table__sort btn caption \\"><span class=\\"base-table__sort-name\\">Mileage</span>
          <svg-icon-stub name=\\"icon-dropdown\\" width=\\"16\\" height=\\"16\\" class=\\"base-table__icon\\"></svg-icon-stub></button></div>
      <div class=\\"base-table__head-data base-table__cell\\"><button type=\\"button\\" class=\\"base-table__sort btn caption \\"><span class=\\"base-table__sort-name\\">Date of purchase</span>
          <svg-icon-stub name=\\"icon-dropdown\\" width=\\"16\\" height=\\"16\\" class=\\"base-table__icon\\"></svg-icon-stub></button></div>
      <div class=\\"base-table__head-data base-table__cell\\"><button type=\\"button\\" class=\\"base-table__sort btn caption \\"><span class=\\"base-table__sort-name\\">Status</span>
          <svg-icon-stub name=\\"icon-dropdown\\" width=\\"16\\" height=\\"16\\" class=\\"base-table__icon\\"></svg-icon-stub></button></div>
      <div class=\\"base-table__head-data base-table__cell\\"><button type=\\"button\\" class=\\"base-table__sort btn caption \\"><span class=\\"base-table__sort-name\\">Trips taken</span>
          <svg-icon-stub name=\\"icon-dropdown\\" width=\\"16\\" height=\\"16\\" class=\\"base-table__icon\\"></svg-icon-stub></button></div>
    </div>
    <div class=\\"base-table__scroller \\">
      <div role=\\"group\\" class=\\"base-table__scroller-wrapper\\" style=\\"padding: 0px 0px 0px;\\">
        <div role=\\"listitem\\" class=\\"base-table__scroller-item\\">
          <div class=\\"base-table__row  \\"><span class=\\"base-table__body-data base-table__cell\\">Niuq1</span><span class=\\"base-table__body-data base-table__cell\\">Mercedes-Benz</span><span class=\\"base-table__body-data base-table__cell\\">C1</span><span class=\\"base-table__body-data base-table__cell\\">Texas, TX</span><span class=\\"base-table__body-data base-table__cell\\">97856</span><span class=\\"base-table__body-data base-table__cell\\">2006-03-03T21:00:00.000Z</span><span class=\\"base-table__body-data base-table__cell\\">fully serviced</span><span class=\\"base-table__body-data base-table__cell\\">541</span></div>
        </div>
        <div role=\\"listitem\\" class=\\"base-table__scroller-item\\">
          <div class=\\"base-table__row  \\"><span class=\\"base-table__body-data base-table__cell\\">bTArp</span><span class=\\"base-table__body-data base-table__cell\\">BMW</span><span class=\\"base-table__body-data base-table__cell\\">A2</span><span class=\\"base-table__body-data base-table__cell\\">Washington DC, DC</span><span class=\\"base-table__body-data base-table__cell\\">24412</span><span class=\\"base-table__body-data base-table__cell\\">2008-04-30T20:00:00.000Z</span><span class=\\"base-table__body-data base-table__cell\\">waiting</span><span class=\\"base-table__body-data base-table__cell\\">761</span></div>
        </div>
        <div role=\\"listitem\\" class=\\"base-table__scroller-item\\">
          <div class=\\"base-table__row  \\"><span class=\\"base-table__body-data base-table__cell\\">bwnVE</span><span class=\\"base-table__body-data base-table__cell\\">Ford</span><span class=\\"base-table__body-data base-table__cell\\">B1</span><span class=\\"base-table__body-data base-table__cell\\">Texas, TX</span><span class=\\"base-table__body-data base-table__cell\\">14475</span><span class=\\"base-table__body-data base-table__cell\\">2002-02-04T21:00:00.000Z</span><span class=\\"base-table__body-data base-table__cell\\">service needed</span><span class=\\"base-table__body-data base-table__cell\\">958</span></div>
        </div>
        <div role=\\"listitem\\" class=\\"base-table__scroller-item\\">
          <div class=\\"base-table__row  \\"><span class=\\"base-table__body-data base-table__cell\\">mMTzR</span><span class=\\"base-table__body-data base-table__cell\\">Mazda</span><span class=\\"base-table__body-data base-table__cell\\">C2</span><span class=\\"base-table__body-data base-table__cell\\">Washington DC, DC</span><span class=\\"base-table__body-data base-table__cell\\">72493</span><span class=\\"base-table__body-data base-table__cell\\">2005-03-06T21:00:00.000Z</span><span class=\\"base-table__body-data base-table__cell\\">service needed</span><span class=\\"base-table__body-data base-table__cell\\">176</span></div>
        </div>
      </div>
      <div style=\\"width: 100%; height: 0px;\\"></div>
    </div>
  </div>
</div>"
`;
